lib/decoding/index.js

```diff
@@ -50,19 +51,22 @@ const bufferToMeasurements = function bufferToMeasurements (buffer, bufferTransf
   }
 
   if (maskLength !== buffer.length) {
-    throw new Error(`incorrect amount of bytes: got ${buffer.length}, should be ${maskLength}`);
+    warnings.push(`incorrect amount of bytes: got ${buffer.length}, should be ${maskLength}`);
+    // throw new Error(`incorrect amount of bytes: got ${buffer.length}, should be ${maskLength}`);
   }
 
   // feed each bufferTransformer element
   for (const mask of bufferTransformer) {
     const maskedBytes = buffer.slice(currByte, currByte + mask.bytes);
 
-    result.push({
-      sensor_id: mask.sensorId,
-      value: mask.transformer(maskedBytes)
-    });
+    if (maskedBytes.length !== 0) {
```

Comment:
Shouldn't this check `maskedBytes.length === mask.bytes` ?

Reply:
Good catch. Resolved with last commit